File: recipe-api/src/main/kotlin/io/bwvolleyball/recipe/domain/Recipe.kt
package io.bwvolleyball.recipe.domain

import mu.KotlinLogging
import javax.annotation.Generated
import javax.persistence.Entity
import javax.persistence.Id

data class Recipe(val id: Long,
                  val name: String,
                  val category: String,
                  val locale: String? = null,
                  val instructions: List<String>,
                  val thumbnail: String,
                  val tags: List<String>,
                  val video: String?,
                  val source: String?,
                  val ingredients: List<Ingredient>,
                  val recipeType: RecipeType)

data class Ingredient(val name: String, val amount: String)

object BlankRecipe {
    val value = Recipe(0L, "", "", null, emptyList(), "", emptyList(), null, null, emptyList(), RecipeType.MEAL)
}

enum class RecipeType {
    MEAL,
    DRINK;

    companion object {

        private val logger = KotlinLogging.logger {  }

        fun recipeType(type: String): RecipeType {
            val upperCased = type.toUpperCase()
            return if (values().map { it.name }.contains(upperCased)) {
                valueOf(type.toUpperCase())
            } else {
                // default to MEAL
                logger.warn("Unable to deterimine RecipeType from {}, defaulting to MEAL", upperCased)
                MEAL
            }
        }
    }
}

File: recipe-organizr-ui/projects/recipe/src/app/recipe/recipe.ts
export class Recipe {
  id: number;
  name: string;
  category: string;
  locale: string;
  instructions: string[];
  thumbnail: string;
  tags: string[];
  video: string;
  source: string;
  ingredients: Ingredient[];
  recipeType: RecipeType;
}

export class Ingredient {
  name: string;
  amount: string;
}

export enum RecipeType {
  MEAL = 'MEAL',
  DRINK = 'DRINK',
}

export class RecipeTypeHelper {
  static fromString(name: string): RecipeType {
    const upperCased = name.toLocaleUpperCase();
    if (upperCased === RecipeType.MEAL.toString()) {
      return RecipeType.MEAL;
    } else if (upperCased === RecipeType.DRINK.toString()) {
      return RecipeType.DRINK;
    } else {
      // this is the default.
      return RecipeType.MEAL;
    }
  }
}
